### PR TITLE
rain rain go away

### PR DESCRIPTION
**Pull request by Axyte**

### CODE REVIEW

File: src/dpp/channel.cpp [hunk 1]
```diff
@@ -78,6 +78,61 @@ std::string channel::get_mention() const {
...
+	return *this;
+}
+
+channel& channel::remove_flag(const channel_flags flag) {
+	this->flags &= ~flag;
+	return *this;
+}
+
+channel& channel::set_nsfw(const bool is_nsfw) {
+	this->flags |= (is_nsfw) ? dpp::c_nsfw : 0;
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:127]: this won't work. you need to use ~ and ^ to invert the bit

File: src/dpp/channel.cpp [hunk 2]
```diff
@@ -78,6 +78,61 @@ std::string channel::get_mention() const {
...
+	this->name = utility::validate(name, 1, 100, "name must be at least 1 character");
+	return *this;
+}
+
+channel& channel::set_topic(const std::string& topic) {
+	this->topic = utility::utf8substr(topic, 0, 1024);
+	return *this;
+}
+
+channel& channel::set_parent_id(const snowflake parent_id) {
```
Comment (braindigitalis) [inline src/dpp/channel.cpp:91]: do we need methods for these,.when we can do c->parent_id = x ?

File: src/dpp/channel.cpp (hunk 1 above)
Comment (Axyte) [reply src/dpp/channel.cpp:127]: Oh thanks, missed that. And did you mean `&` instead of `^`?

File: src/dpp/channel.cpp (hunk 2 above)
Comment (Axyte) [reply src/dpp/channel.cpp:91]: I mean that logic can be applied to every other class but we still have setters, don't we?

**Pull request merged**